File: src/app/services/IMetricService.ts
import { Metric } from '../model/Metric';
import { IHeartRateSensorService } from './IHeartRateSensorService';
import { ILocationService } from './ILocationService';
import { IPowerMeterService } from './IPowerMeterService';
import { ISpeedSensorService } from './ISpeedSensorService';


export interface IMetricService {
    getLocationService(): ILocationService;
    getHeartRateSensorService(): IHeartRateSensorService;
    getPowerMeterService(): IPowerMeterService;
    getSpeedSensorService(): ISpeedSensorService;
    getByName(name: string): Metric | null;
    getNames(): string[];
}

File: android/app/src/main/java/com/openbikecomputer/service/IMetricService.java
package com.openbikecomputer.service;

/**
 * Subset of the metric service required by {@code Metric} implementations to
 * reach the underlying sensor services. Ported from {@code IMetricService.ts}.
 */
public interface IMetricService {
    ILocationService getLocationService();

    IHeartRateSensorService getHeartRateSensorService();

    IPowerMeterService getPowerMeterService();

    ISpeedSensorService getSpeedSensorService();
}
